src/services/dataService.js

```diff
@@ -0,0 +1,88 @@
+import * as axios from 'axios';
+import {
+  API_AUTH_URL,
+  API_BASE_URL,
+  DATA_TYPES,
+  API_SEARCH_LOCATION_DATA_TYPE,
+  API_GET_CURRENT_WEATHER_DATA_TYPE,
+  API_GET_FORECAST_DATA_TYPE
+} from '../constants/constants';
+
+const instance = axios.create({
+  withCredentials: true,
+  baseURL: API_BASE_URL
+});
+
+export const weatherAPI = {
+  token: null,
+  getAuthHeaders() {
+    return {
+      headers: {
+        Authorization: `Bearer ${this.token}`
+      }
+    };
+  },
+  async getToken(user, password) {
+    const body = {
+      user,
+      password
+    };
+    try {
+      const response = await instance.post(API_AUTH_URL, body, {
+        headers: {
+          'Content-Type': 'text/plain'
+        }
+      });
+      this.token = response.data.access_token;
+    } catch (error) {
+      console.log(error);
+      this.token = null;
+    }
+
+    return this.token;
+  },
+
+  async getData(dataType, param) {
+    const { url, dataKey } = DATA_TYPES[dataType];
+    try {
+      const response = await instance.get(url + param, this.getAuthHeaders());
+      return response.data[dataKey];
+    } catch (error) {
+      console.log(error);
+      return null;
+    }
+  },
+
+  async searchLocation(query) {
+    const { url, dataKey } = DATA_TYPES[API_SEARCH_LOCATION_DATA_TYPE];
+    try {
+      const response = await instance.get(url + query, this.getAuthHeaders());
+      return response.data[dataKey];
+    } catch (error) {
+      console.log(error);
+      return null;
+    }
+  },
+
+  async getCurrentWeather(locationId) {
+    const { url, dataKey } = DATA_TYPES[API_GET_CURRENT_WEATHER_DATA_TYPE];
+    try {
+      const response = await instance.get(url + locationId, this.getAuthHeaders());
```

Comment:
why concatenate, could you use params object instead?
```
axios.get('/user', {
    params: {
      ID: 12345
    }
})
```

taken from axios docs

Reply:
did not know about this syntax, thanks. But in this case it's not suitable because I need for example to get '/url/locationId' instead of '/url?locationId=locationIdValue'.
So I think it's better to use manually concatenation because it gives flexibility.

Reply:
Ok, got it. Could you use ES6 strings instead? It makes it a bit more readable and more consistent with the rest parts of the project